File: src/types/system-skill.ts
import type { MarketplaceInstallMetadata } from "./marketplace-install-metadata";
import type { SystemSkillSource } from "./system-skill-source";

export type SystemSkill = {
  id: string;
  slug: string;
  name: string;
  summary: string;
  manifestPath: string;
  rootPath: string;
  source: SystemSkillSource | string;
  marketplaceInstall?: MarketplaceInstallMetadata | null;
};

File: src-tauri/src/models/system_skill.rs
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct SystemSkill {
    pub id: String,
    pub slug: String,
    pub name: String,
    pub summary: String,
    pub manifest_path: String,
    pub root_path: String,
    pub source: String,
}
